content/blog/coredns-in-kubernetes-via-kubeadm.md

```diff
@@ -0,0 +1,298 @@
+coredns   1                     1                       1                         1           4h
+~~~
+
+Check the `Configmap`:
+~~~ text
+# kubectl -n kube-system get configmap coredns -oyaml
+apiVersion: v1
+data:
+  Corefile: |
+    .:53 {
+        errors
+        health
+        kubernetes cluster.local 10.96.0.0/12 {
+           pods insecure
+           upstream /etc/resolv.conf
+        }
+        prometheus :9153
+        proxy . /etc/resolv.conf
+        cache 30
+    }
+kind: ConfigMap
+metadata:
+  creationTimestamp: 2017-12-21T12:55:15Z
+  name: coredns
+  namespace: kube-system
+  resourceVersion: "161"
+  selfLink: /api/v1/namespaces/kube-system/configmaps/coredns
+  uid: 30bf0882-e64e-11e7-baf6-0cc47a8055d6
+~~~
+
+>It is important to note that when switching from kube-dns to CoreDNS, the configurations that come with kube-dns (stubzones, federations...) will no longer exist and will switch to a default configuration in CoreDNS.
+>The translation of the configurations from kube-dns shall be supported from the upcoming version of Kubernetes (v1.10) where CoreDNS will be Beta.
+
+We can check if CoreDNS is functioning normally through a few basic `dig` commands:
+
+~~~ text
+# dig @10.96.0.10 kubernetes.default.svc.cluster.local
+
+; <<>> DiG 9.10.3-P4-Ubuntu <<>> @10.96.0.10 kubernetes.default.svc.cluster.local
+; (1 server found)
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 58006
+;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1
+
+;; OPT PSEUDOSECTION:
+; EDNS: version: 0, flags:; udp: 4096
+;; QUESTION SECTION:
+;kubernetes.default.svc.cluster.local. IN A
+
+;; ANSWER SECTION:
+kubernetes.default.svc.cluster.local. 5	IN A	10.96.0.1
+
+;; Query time: 0 msec
+;; SERVER: 10.96.0.10#53(10.96.0.10)
+;; WHEN: Thu Dec 21 18:10:49 UTC 2017
+;; MSG SIZE  rcvd: 81
+
+
+# dig @10.96.0.10 ptr 1.0.96.10.in-addr.arpa.
+
+; <<>> DiG 9.10.3-P4-Ubuntu <<>> @10.96.0.10 ptr 1.0.96.10.in-addr.arpa.
+; (1 server found)
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 30847
+;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1
+
+;; OPT PSEUDOSECTION:
+; EDNS: version: 0, flags:; udp: 4096
+;; QUESTION SECTION:
+;1.0.96.10.in-addr.arpa.		IN	PTR
+
+;; ANSWER SECTION:
+1.0.96.10.in-addr.arpa.	5	IN	PTR	kubernetes.default.svc.cluster.local.
+
+;; Query time: 0 msec
+;; SERVER: 10.96.0.10#53(10.96.0.10)
+;; WHEN: Thu Dec 21 18:09:10 UTC 2017
+;; MSG SIZE  rcvd: 101
+
+~~~
+
+Below is the `log` for some of the basic queries.
+~~~ text
+# kubectl -n kube-system logs coredns-546545bc84-p4x7k
+.:53
+CoreDNS-1.0.1
+linux/amd64, go1.9.1, a04eeb9c
+2017/12/21 18:01:35 [INFO] CoreDNS-1.0.1
+2017/12/21 18:01:35 [INFO] linux/amd64, go1.9.1, a04eeb9c
+10.32.0.1 - [21/Dec/2017:18:06:34 +0000] "A IN kube-dns.kube-system.svc.cluster.local. udp 68 false 4096" NOERROR qr,aa,rd,ra 84 104.117µs
```

Comment:
In the latest default deployment config, we'll no longer log every query received.  This section will confuse those who have the newer config.